### PR TITLE
v3 data availability bug

### PR DESCRIPTION
**Pull request by cbergz**

### CODE REVIEW

File: src/tx-builder/services/DydxToken.ts [hunk 1]
```diff
@@ -1,370 +1,370 @@
...
-      .sub(parseEther(LOCKED_ALLOCATION))
-      .sub(rewardsTreasuryBalance)
-      .sub(rewardsTreasuryVesterBalance)
-      .sub(communityTreasuryBalance)
-      .sub(communityTreasuryVesterBalance);
-
-    return formatEther(circulatingSupplyWei);
-  };
-}
+// import { formatEther, parseEther } from '@ethersproject/units';
```
Comment (lucas-dydx) [inline src/tx-builder/services/DydxToken.ts:1]: For my understanding why did this file have to be commented out?

File: tasks/deployment/v3-data-availability-proposal.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+import mainnetAddresses from '../../src/deployed-addresses/mainnet.json';
+import { hardhatTask } from '../../src/hre';
+import { DIP_22_IPFS_HASH } from '../../src/lib/constants';
+import { createV3DataAvailabilityProposal } from '../../src/migrations/v3-data-availability-proposal';
+
+hardhatTask('deploy:v3-data-availability-proposal', 'Create proposal to fix data availability bug on v3 perp contract.')
+  .addParam('proposalIpfsHashHex', 'IPFS hash for the uploaded DIP describing the proposal', DIP_22_IPFS_HASH, types.string)
+  .addParam('governorAddress', 'Address of the deployed DydxGovernor contract', mainnetAddresses.governor, types.string)
+  .addParam('starkwarePriorityAddress', 'Address of the deployed short timelock Executor contract', mainnetAddresses.starkwarePriorityTimelock, types.string)
```
Comment (lucas-dydx) [inline tasks/deployment/v3-data-availability-proposal.ts:11]: Nit: mind updating the description to mention this is the deployed starkware priority executor contract?

File: tasks/deployment/v3-data-availability-proposal.ts [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+  .addParam('starkPerpetualAddress', 'Address of the deployed liquidity module contract', mainnetAddresses.starkPerpetual, types.string)
```
Comment (lucas-dydx) [inline tasks/deployment/v3-data-availability-proposal.ts:12]: Nit: mind updating this description as well?

File: test/migrations/v3-data-availability-proposal.ts [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+}: {
+  starkwarePriorityAddress: string,
+  starkPerpetualAddress: string,
+}) {
+  //const deployConfig = getDeployConfig();
+  const starkwarePrioritySigner = await impersonateAndFundAccount(starkwarePriorityAddress);
+  const starkPerpetual = new MockStarkPerpetual__factory(starkwarePrioritySigner).attach(starkPerpetualAddress);
+
+  await starkPerpetual.mainAcceptGovernance();
+  //await waitForTx(await starkPerpetual.registerGlobalConfigurationChange(deployConfig.STARK_PERPETUAL_CONFIG_HASH));
```
Comment (lucas-dydx) [inline test/migrations/v3-data-availability-proposal.ts:31]: For my understanding why are these lines commented out?

File: test/misc/v3-data-availability-proposal.spec.ts [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+
+describeContract('update the funding rate config hash', init, (ctx: TestContext) => {
+
+  it.only('Proposal IPFS hash is correct', async () => {
+    const V3_DATA_AVAILABILITY_PROPOSAL_ID = 21;
+    const proposal = await ctx.governor.getProposalById(V3_DATA_AVAILABILITY_PROPOSAL_ID);
+    expect("0x0000000000000000000000000000000000000000000000000000000000000000").to.equal(DIP_22_IPFS_HASH);
+  });
+
+  it.only('Global configuration hash has been updated', async () => {
```
Comment (lucas-dydx) [inline test/misc/v3-data-availability-proposal.spec.ts:18]: Nit: can we remove the .only?

File: hardhat.config.ts [hunk 6]
```diff
@@ -61,11 +61,14 @@ function getRemoteNetworkConfig(
 
 function getHardhatConfig(): HardhatNetworkUserConfig {
   const networkConfig: HardhatNetworkUserConfig = {
-    hardfork: 'berlin',
-    blockGasLimit: 15000000,
+    //hardfork: 'berlin',
```
Comment (lucas-dydx) [inline hardhat.config.ts:64]: Nit: if not using this can we delete it?

File: src/migrations/v3-data-availability-proposal.ts [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+        'upgradeTo(address,bytes,bool)',
+    ],
+    [
+        hre.ethers.utils.defaultAbiCoder.encode(
+            [],
+            [],
+        ),
+        hre.ethers.utils.defaultAbiCoder.encode(
+            ['bytes32'],
+            [deployConfig.STARK_PERPETUAL_CONFIG_HASH],
```
Comment (lucas-dydx) [inline src/migrations/v3-data-availability-proposal.ts:53]: Just to be safe, what do you think about throwing an error if this string is empty?

File: hardhat.config.ts (hunk 6 above)
Comment (cbergz) [reply hardhat.config.ts:64]: sorry yeah I was playing with the config to see why my local test was failing

File: src/tx-builder/services/DydxToken.ts (hunk 1 above)
Comment (cbergz) [reply src/tx-builder/services/DydxToken.ts:1]: for some reason this crashes everything for me - haven't been able to figure out how to resolve

Comment (cbergz) [reply src/tx-builder/services/DydxToken.ts:1]: uncommented now

File: test/migrations/v3-data-availability-proposal.ts (hunk 4 above)
Comment (cbergz) [reply test/migrations/v3-data-availability-proposal.ts:31]: sorry I was playing around with testing after hitting some issues on npm run test

**Pull request merged**